docs/contribute/dev-setup.md

```diff
@@ -0,0 +1,122 @@
+Thank you for considering to contribute to this project. This page shows you how to setup your local
+development environment.
+
+### [Optional] Install Docker
+
+If you have Docker installed, you can skip this step.
+
+To install Docker, follow the instructions on the [official website](https://docs.docker.com/get-docker/).
+
+### [Optional] Install Docker Compose
+
+If you have Docker Compose installed, you can skip this step.
+After installing Docker, you can install Docker Compose by following the instructions on the [official website](https://docs.docker.com/compose/install/).
+
+### Clone the repository
+
+Clone the [HCD-AI](https://github.com/robellmh/HCD-AI) repository.
+
+```shell
+git clone git@github.com:robellmh/HCD-AI.git
+```
+
+### Create a virtual environment
+
+Navigate to the root of the project and create a virtual environment.
+if you are happy using `conda`, you can create it using the following command:
+
+```shell
+make fresh-env
+```
+
+This will install the dependencies in `requirements.txt` and `requirements-dev.txt` in a new conda environment called `hew-ai`.
+
+Now active the environment:
+
+```shell
+conda activate hew-ai
+```
+
+### Setup pre-commit hooks
+
+To setup pre-commit hooks, run the following command:
+
+```shell
+pre-commit install
+```
+
+### Setup environment variables
+
+
+Navigate to the `deployment/docker-compose/` subfolder.
+
+```shell
+cd deployment/docker-compose/
+```
+
+Copy `template.*.env` files to `.*.env`:
+
+```shell
+cp template.base.env .base.env
+cp template.backend.env .backend.env
+```
+
+Edit the `.base.env` and `.backend.env` files to set the environment variables.
```

Comment:
I would add something here reminding people to remove the `#pragma` comments as they can get in the way sometimes

Reply:
Really? Why?
I'll just remove it since we dont run the secrets checker pre-commit here anyway